File: crates/klyron_bundler/bindings/ts/benchmark.ts
import { BundlerClient } from './client';

export function benchBundlerVersion(iterations: number): number {
  const client = new BundlerClient();
  const start = process.hrtime.bigint();
  for (let i = 0; i < iterations; i++) { client.version(); }
  return Number(process.hrtime.bigint() - start) / 1e6;
}

File: crates/klyron_bundler/bindings/rust/benchmark.rs
#![cfg(test)]

#[test]
fn bench_version() {
    let client = crate::client::BundlerClient::new();
    let start = std::time::Instant::now();
    for _ in 0..1000 { let _ = client.version(); }
    println!("version() x1000: {:?}", start.elapsed());
}
